specification/communication/data-plane/role/preview/2022-01-01-preview/CommunicationRole.json

```diff
@@ -0,0 +1,439 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationRoleClient",
+    "description": "Azure Communication Role service",
+    "version": "2022-01-01-preview"
+  },
+  "paths": {
+    "/roles": {
+      "post": {
```

Comment:
If we have to stick with `post` in the near future because of unresolved privacy discussions, will we support the `Repeatability-Request-ID` header to make this route idempotent?

We can also consider to make `displayName` the id in the route and use PUT. Which consequences would this have in the client SDKs where we probably still need the service-generated GUID?